bite-size-notes/Week3-c_TrajectoryOptimization_Devansh.md

```diff
@@ -0,0 +1,133 @@
+3. Sometimes, one will have to solve the problem over and over again.
+
+### Continuous Time Dynamics
+
+$\min_{x_(0)...x_t+1, u_(0)...u_(t)} J_termial(x_(t+1)) + \int_{t=0}^{T_f} J(x_(t), u_(t), t)$$
+
+$$\text{subject to} \quad x_{k+1} = f(x_(t), u_(t), t), \quad k=0,...K$$
+
+$$ \quad \quad \quad x_t \in X, \quad u_(t) \in U_(t), \quad x_(0) = x_current or u_(t) \in U_(t)(t_k)$$ 
+
+## Convex Formulation for Linear Systems
+
+### Continuous-Time Dynamics
+For linear systems with continuous-time dynamics, the trajectory optimization problem can be formulated as a convex optimization problem. Consider a continuous-time linear system with state $( x(t) )$ and control input $( u(t) )$, governed by the dynamics:
+$$\dot{x}(t) = Ax(t) + Bu(t)$$
+
+### Discrete-Time Dynamics
+For discrete-time linear systems, the trajectory optimization problem is naturally formulated as a convex optimization problem. Consider a discrete-time linear system with state $( x_k )$ at time step $( k )$ and control input $( u_k )$, governed by the dynamics:
+$$x_{k+1} = Ax_k + Bu_k$$
+
+## Detailed Example: Continuous-Time Dynamics
+Consider a simple continuous-time linear system with state $( x(t) = [x_1(t), x_2(t)]^T )$ representing position and velocity, and control input $u(t) = u_1(t)$ representing acceleration. Let $( A )$ and $( B )$ be given matrices defining the system dynamics.
+
+The cost function could be a quadratic form:
+$$J = \int_{t_0}^{t_f} \left( x^T Q x + u^T R u \right) \, dt $$
+
+Where (Q), (R) are positive definite weighting matrices.
+This problem can be further solved using convex optimization techniques such as QP or other solvers capable of handling convex problems.
+
+## Detailed Example: Discrete-Time Dynamics
+Consider the same linear system with discrete-time dynamics:
+$$x_{k+1} = Ax_k + Bu_k$$
+
+We want to minimize the cost function:
+$$J = \sum_{k=0}^{N-1} \left( x_k^T Q x_k + u_k^T R u_k \right) + x_N^T Q_f x_N$$
+
+Where (Q), (R) and $(Q_f)$ are positive definite weighting matrices.
+This problem can be further solved using convex optimization techniques such as QP or other solvers capable of handling convex problems. 
+
+## CONCLUSION
+
+We understand the importance of CLF's and CBF's how they are used in Trajectroy optimization and also about the disjoint sets. Moreover, the problem of optimization stated above stands true only for Linear systems and also which are convex in nature. For non-convex problems, the system is first converted to convex-type and then solved further. 
+We have discussed the Direct Transcription method for Trajectory Optimization, but there are other methods such as shooting method which gives a general idea of reaching a goal within a given time interval and also tells us if the trajectory is dynamically feasible or not. We have not quite discussed the QP, quadratic programming part but its a part of Trajectory optimization which can be discussed further. 
+
```

Comment:
Overall I think these bite-sized-notes are concise and easy to read except for some minor typos and Latex formatting issues.

Reply:
Maybe consider adding pictures too.

Reply:
https://www.youtube.com/watch?v=_Tkn_Hzo4AA

I think this is a nice video on the introduction to CLFs and CBFs we could link on these notes.

Reply:
Thanks for this video. I have removed all the typos and added the hyperlink in the beginning. Also, I could not think of any pictures for this extract, and it would be an overlap with the person who is explaining the CLF's and CBF's.